### ISSUE TITLE
fish_color_keyword doesn't affect keyword colors

### ISSUE THREAD

**Issue by EmilySeville7cfg** [fish-shell/fish-shell#8517]
Fish version: 3.3.1-682-g0cd3ed4b8-dirty (built from source)
OS version: Linux Mint 20.2 Cinnamon

It is expected `if` keyword to be bold blue color on the screenshot but not bold cyan:

![image](https://user-images.githubusercontent.com/42812113/144367742-91707e59-ae8d-4eab-b71c-092557d10070.png)

**Comment by faho** [comment]
No, this is expected. Just `if` or `if --help` is still treated as a builtin by the parser and will print the help.

**Comment by EmilySeville7cfg** [comment]
The documentation confuses me if it's true:

> fish_color_keyword: keywords like if - this falls back on the command color if unset

If I am right, it says Fish will use `fish_color_command` for keywords like `if` if `fish_color_keyword` is unset.

**Comment by floam** [comment]
`fish_color_keyword` does affect `if`. The thing is though, however, that there is a bit of a quirk in that it doesn't really become a keyword as far as the parser is concerned until after your start entering the condition. There is sort of a hack necessary in order for it to be able to take `--help`. `if --help` is actually a builtin under the hood.

**Comment by floam** [comment]
<img width="281" alt="image" src="https://user-images.githubusercontent.com/291142/144376424-b64e3783-9fa4-430f-af1c-52bd84c705d3.png">

Now maybe, this is confusing now that we have this syntax highlighting, and we should try to hide this implementation detail from the user and force the these shadowed keywords to highlight sooner, and mark this as an enhancement request? @faho?

**Comment by floam** [comment]
specifically, I can see a decent argument for the `if ` case.

**Comment by faho** [comment]
Yeah, that makes sense. Probably needs an additional parser flag or something.
